### ISSUE TITLE
Inconsistent behavior of formatOnType for multi-line with different line break types

### ISSUE THREAD

**Issue by StellaHuang95** [microsoft/pylance-release#4456]
## Environment data

-   Language Server version: v2023.5.51

## Code Snippet

```python
def foo():
    return 1 * \
           2 # <- Hit Enter here

# This is the last line of the file
```

## Repro Steps

1. Create a python file with code snippet above in VSCode
2. Set `editor.formatOnType` to `true`, and set `editor.autoIndent` to `none`
3. Set line break type to `CRLF`, hit Enter after `2` and observe the auto-indent behavior
4. Set line break type to `LF`, hit Enter after `2` and observe the auto-indent behavior

## Expected behavior

Both of they should behave the same and not indent.

## Actual behavior

With line ending set to LF:

![formatOnType LF](https://github.com/microsoft/pylance-release/assets/100439259/64bd5e98-b55c-4450-a15a-cd4e7e35bbe5)

With line ending set to CRLF:

![formatOnType CRLF](https://github.com/microsoft/pylance-release/assets/100439259/2ef08eaf-ec8a-416a-bbec-cdc33367c5e7)

**Comment by StellaHuang95** [comment]
Another issue found in ptvs: https://github.com/microsoft/PTVS/issues/7649, also repros in vscode with `editor.formatOnType` set to `true` and "editor.autoIndent" set to `none`,

The auto-indent doesn't apply to comments.
![autoindent](https://github.com/microsoft/pylance-release/assets/100439259/6a87c8cb-b9b8-49e9-8af2-851a235af8b0)

**Comment by rchiodo** [comment]
We might start checking 'editor.autoIndent' to set our indent amount. So 'editor.autoIndent' of none should not indent at all, ever.

**Comment by debonte** [comment]
In the first scenario ("Enter after 2 in multiline return statement"), the difference in behavior is due to the following code in `_formatOnTypeEnter`:

```typescript
        // Reduce offset by one to properly handle case where user uses Enter to split a line
        // (i.e. the new line has content to the right of the cursor). Otherwise the token at the
        // cursor position can impact the indent calculation.
        const indentCalculationOffset = offset > 0 ? offset - 1 : offset;
```

With `LF` line endings, if we don't subtract one from `offset`, the behavior is the same as `CRLF`.

I'm not sure if we should change our behavior based on the value of `editor.autoIndent`. That setting controls what the editor does before our auto-indent code is run and seems somewhat independent of what we're doing. For example, users might want to use `editor.autoIndent` = `full` but still turn off our auto-indent functionality.

Also, providing no indent when `editor.autoIndent` is `none` won't solve the second scenario ("Enter after comment") in VS. I think this falls into the class of auto-indent bugs where the editor's indent logic is covering for an omission in our auto-indent logic. When it's `none` (or in VS), the editor doesn't provide the indent after a comment and we don't either.

**Comment by StellaHuang95** [comment]
Sorry for putting multiple issues in one ticket, I've created a [new issue](https://github.com/microsoft/pylance-release/issues/4763) to track the second scenario as we're getting some more complaints from VS users.

**Comment by rchiodo** [comment]
This issue has been fixed in prerelease version 2023.9.21, which we've just released. You can find the changelog here: [CHANGELOG.md](https://github.com/microsoft/pylance-release/blob/main/CHANGELOG.md#2023921-20-September-2023-PreRelease)
